src/example-types/esTwoLevelAggregation.js

```diff
@@ -72,7 +94,15 @@ module.exports = {
                 key: bucket.key || key,
                 doc_count: bucket.doc_count,
               },
-              bucket.twoLevelAgg
+              bucket.twoLevelAgg ||
+                F.reduceIndexed(
+                  (obj, data, key) =>
+                    _.startsWith('twoLevelAgg_', key)
+                      ? _.extend({ [_.last(key.split('_'))]: data.value }, obj)
+                      : obj,
+                  {},
+                  bucket
```

Comment:
this reduceIndexed could be replaced with something that does less juggling:

```js
_.flow(
  _.mapKeys(_.replace('twoLevelAgg_', '')),
  _.mapValues('value')
)
```

Reply:
Once you do that, to handle the unified scenario (where stats is requested as `twoLevelAgg_stats`), you could replace the mapValues with a getOrReturn, and then just have a list known metrics that get merged onto the result (so, for example, `stats` would be merged onto `bucket` and deleted off `bucket`).

Another simplification you could make is to just do away with the whole `twoLevelAgg_` prefix across this whole thing. It's just extra complication for not much gain.